lms/djangoapps/discussion_api/api.py

```diff
@@ -415,7 +415,7 @@ def get_comment_list(request, thread_id, endorsed, page, page_size, mark_as_read
         raise Http404
     num_pages = (resp_total + page_size - 1) / page_size if resp_total else 1
 
-    results = [CommentSerializer(response, context=context).data for response in responses]
+    results = [CommentSerializer(response, remove_fields=["children"], context=context).data for response in responses]
```

Comment:
Having said that, there should be a way for the discussion_api to indicate to the ruby code to not send the children.  Whether that's done with the `recursive` parameter, or another parameter, is an implementation detail.  If `recursive` is currently not used, then we can use it.

@wajeeha-khalid Can you become familiar enough with Ruby to update it to not return the children?  For now, it can just exclude the children in the return payload.  The next step would be to update `merge_response_content` to not actually access the children.